### ISSUE TITLE
Add code folding to the docs

### ISSUE THREAD

**Issue by nettoyoussef** [emacs-evil/evil#1380]
# Issue type
- Enhancement request

# Environment
Not relevant.

# Problem
Currently there is no information about folding or evil-fold-list in the docs (e.g, [here](https://readthedocs.org/projects/evil/downloads/pdf/latest/) or [here](https://evil.readthedocs.io)).

Per this [reddit thread](https://www.reddit.com/r/emacs/comments/6fmpwb/evil_and_builtin_folding/) it is possible to select a backend by editing `evil-fold-list`, such as this [example](https://www.reddit.com/r/emacs/comments/3nnbkf/customize_the_evil_mode_folding_add_your_own/):

```lisp
(defvar evil-fold-list
  `(((hs-minor-mode)
     :open-all   hs-show-all
     :close-all  hs-hide-all
     :toggle     hs-toggle-hiding
     :open       hs-show-block
     :open-rec   nil
     :close      hs-hide-block)
   )
)
```

or to add a hook to enable folding with a particular package (one of vdiff-mode, vdiff-3way-mode, hs-minor-mode, hide-ifdef-mode, outline-mode, origami-mode) in a [particular mode](https://www.reddit.com/r/emacs/comments/4h1f2d/question_how_can_i_enable_hideshow_for_all_its/):

```lisp
(use-package evil
  :ensure t
  :hook
  ('prog-mode-hook . 'hs-minor-mode)
  :config
  (evil-mode 1)
)
```

It would be nice to have this in the documentation somewhere.

**Comment by jvasile** [comment]
I agree more information about folding would be appropriate.  Or any, really.  Even emacs wiki seems to be lacking on this topic.  In addition to the above, a simple introduction might be useful.  Perhaps something like this:

Evil includes document folding commands:

    evil-toggle-fold
    evil-open-folds
    evil-close-folds
    evil-open-fold
    evil-open-fold-rec
    evil-close-fold

Each of these matches one of a set of folding commands, depending on which minor mode is active.  For many major modes, turning on origami-mode or outline-minor-mode will enable folding via zc, zo, zm, and zr.  For more information on associating new modes with the evil folding commands, see documentation for the `evil-fold-list` variable.
